app/src/main/java/com/jakdor/apapp/ui/registration/RegistrationFragment.kt

```diff
@@ -0,0 +1,279 @@
+            }
+
+            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {
+
+            }
+
+            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+                val email: String = email_editText.text.toString()
+
+                when {
+                    viewModel?.validateEmail(email) != null -> {
+                        email_editText.error = viewModel?.validateEmail(email)
+                        isEmailCorrect = false
+                    }
+                    viewModel?.isEmptyValidation(email)!=null -> {
+                        email_editText.error = viewModel?.isEmptyValidation(email)
+                        isEmailCorrect = false
+                    }
+                    else -> {
+                        email_editText.error = null
+                        isEmailCorrect = true
+                    }
+                }
+            }
+        })
+        name_editText.addTextChangedListener(object: TextWatcher{
+            override fun afterTextChanged(s: Editable?) {
+                register_button.isEnabled = (isPasswordCorrect && isRePasswordCorrect && isEmailCorrect
+                        && isNameNotEmpty && isSurnameNotEmpty && isLoginCorrect)
+            }
+
+            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {}
+
+            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+                val name: String = name_editText.text.toString()
+
+                if(viewModel?.isEmptyValidation(name) != null){
+                    name_editText.error = viewModel?.isEmptyValidation(name)
+                    isNameNotEmpty = false
+                }else{
+                    name_editText.error = null
+                    isNameNotEmpty = true
+                }
+            }
+        })
+        surname_editText.addTextChangedListener(object: TextWatcher{
+            override fun afterTextChanged(s: Editable?) {
+                register_button.isEnabled = (isPasswordCorrect && isRePasswordCorrect && isEmailCorrect
+                        && isNameNotEmpty && isSurnameNotEmpty && isLoginCorrect)
+            }
+
+            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {}
+
+            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+
+                val surname: String = surname_editText.text.toString()
+
+                if(viewModel?.isEmptyValidation(surname) != null){
+                    surname_editText.error = viewModel?.isEmptyValidation(surname)
+                    isSurnameNotEmpty = false
+                }else{
+                    surname_editText.error = null
+                    isSurnameNotEmpty = true
+                }
+            }
+        })
+        login_editText.addTextChangedListener(object: TextWatcher{
```

Comment:
Z tego co widzę to chcesz we ViewModelu sprawdzać czy istnieje już na bazie. Zastanów się dla czego sprawdzanie tego co każdą najmniejszą zmianę tekstu to głupi pomysł.